lib/msal-angular/docs/initialization-2.x.md

```diff
@@ -0,0 +1,163 @@
+            auth: {
+                clientId: "Your client ID"
+            }
+        })
+    ]
+})
+export class AppModule {}
+```
+
+## Secure the routes in your application
+
+You can add authentication to secure specific routes in your application by just adding `canActivate : [MsalGuard]` to your route definition. It can be added at the parent or child routes.
+
+```js
+{
+    path: 'product',
+    component: ProductComponent,
+    canActivate: [MsalGuard],
+    children: [
+        {
+            path: 'detail/:id',
+            component: ProductDetailComponent
+        }
+    ]
+}, {
+    path: 'myProfile',
+    component: MsGraphComponent,
+    canActivate: [MsalGuard]
+},
+```
+
+When a user visits these routes, the library will prompt the user to authenticate.
+
+## Get tokens for Web API calls
+
+MSAL Angular allows you to add an Http interceptor (`MsalInterceptor`) in your `app.module.ts` as follows. MsalInterceptor will obtain tokens and add them to all your Http requests in API calls based on the `protectedResourceMap`.
+
+```js
+@NgModule({
+    imports: [
+        MsalModule.forRoot({
+            auth: {
+                clientId: "Your client ID"
+            }
+        }, {
+        //MsalGuardConfiguation here
+        }, {
+            protectedResourceMap: new Map([
+                ['https://graph.microsoft.com/v1.0/me', ['user.read']],
+                ['https://api.myapplication.com/users/*', ['customscope.read']]
+            ])
+        })
+    ],
+    providers: [
+        ProductService, 
+        {
+            provide: HTTP_INTERCEPTORS,
+            useClass: MsalInterceptor,
+            multi: true
+        }
+    ]
+})
+export class AppModule {}
+```
+
+Using MsalInterceptor is optional and you can write your own interceptor if you choose to. Alternatively, you can also explicitly acquire tokens using the acquireToken APIs.
+
+As of `@azure/msal-angular@2`protectedResourceMap` supports wildcard patterns that are supported by [minimatch](https://github.com/isaacs/minimatch). `unprotectedResources` is deprecated and no longer an option fpr configuration. 
+
+**Note:** When using wildcards, if multiple matching entries are found in the `protectedResourceMap`, the first match found will be used (based on the order of the `protectedResourceMap`).
+
+## Subscribe to event callbacks
+
+MSAL wrapper provides below callbacks for various operations. For all callbacks, you need to inject BroadcastService as a dependency in your component/service and also implement a `handleRedirectObservable`:
+
+```js
+this.authService.handleRedirectObservable().subscribe({
+    next: (result) => // do something here
+});
+```
+
+### 1. How to subscribe to events
+
+```js
+this.msalBroadcastService.msalSubject$
+    .pipe(
+        filter((msg: EventMessage) => msg.eventType === EventType.LOGIN_SUCCESS)
+    )
+    .subscribe((result) => {
+        // do something here
+    });
+```
+
+### 2. List of available events
```

Comment:
I think just linking to the existing doc from browser is simpler.

Reply:
Makes sense. Done.